weasyprint/draw/text.py

```diff
@@ -176,10 +177,21 @@ def draw_first_line(stream, textbox, text_overflow, block_ellipsis, matrix):
             glyph_info = glyphs[i]
             glyph = glyph_info.glyph
             width = glyph_info.geometry.width
-            if (glyph == pango.PANGO_GLYPH_EMPTY or
-                    glyph & pango.PANGO_GLYPH_UNKNOWN_FLAG):
+            if glyph == pango.PANGO_GLYPH_EMPTY:
+                # Display empty glyph, such as space.
                 string += f'>{-width / font_size}<'
                 continue
+            elif glyph & pango.PANGO_GLYPH_UNKNOWN_FLAG:
+                # Display tofu for missing glyph.
+                string += f'{0:02x}' if font.bitmap else f'{0:04x}'
+                string += f'>{-width / font_size}<'
+                utf8_slice = slice(*sorted(utf8_positions[i:i+2]))
+                characters = utf8_text[utf8_slice].decode()
+                codepoints = ', '.join(f'U+{ord(char)}:04x' for char in characters)
```

Comment:
The format is wrong: https://github.com/unicode-org/core-spec/actions/runs/16891391687/job/47851784654#step:13:29

The mistakes like “U+3164:04x” should be “U+0C5C” (not “U+0c5c” either).

```suggestion
                codepoints = ', '.join(f'U+{ord(char):04X}' for char in characters)
```

Reply:
Sorry, my mistake.